unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -99,6 +99,18 @@ public static IntPtr object_isinst([NativeCallbackType("MonoObject*")] IntPtr ob
     public static IntPtr object_get_class([NativeCallbackType("MonoObject*")] IntPtr obj)
         => obj.ToManagedRepresentation().TypeHandleIntPtr();
 
+    [return: NativeCallbackType("MonoObject*")]
+    public static IntPtr array_new([NativeCallbackType("MonoObject*")] IntPtr domain, [NativeCallbackType("MonoClass*")] IntPtr klass, [NativeWrapperType("guint32")] int n)
+        => Array.CreateInstance(klass.TypeFromHandleIntPtr(), n).ToNativeRepresentation();
```

Comment:
This is where performance may start to come into play. I think it's fine to proceed with correctness first.

Questions worth considering
1. How does this implementation perform compared to the previous C++ code? 
2. To what extent does Unity use this API and what performance expectations does it have?
3. What are the options to improve this performance? Maybe a `Dictionary<IntPtr, Func<object>>` that would be type specific functions to allocate an object of a given type.

Reply:
As I add apis, I think it would make more sense to handle performance review separately.   At least when a trivial implementation is possible.  The advantages being

*  By getting an implementation in, we unblock other efforts such as if/how the testing approach will evolve.
*  Performance testing will require additional setup.  W/e that setup is, I think it would be nice to have a few apis to look at rather than look at 1 and then switch off of performance.
*  Depending on how the testing approach evolves, maybe that would impact the approach we take to performance testing.

I made https://jira.unity3d.com/browse/VM-1974 to start capturing the apis we should come back to.

Reply:
Agreed and thanks!